hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -46,286 +79,392 @@ def find_target(target_name = ''):
-    for key, val in pinName_dict.items():
+    for key, val in pin_name_dict.items():
         if re.match("^ARDUINO_UNO_[AD]\d+$", key):
-            if val == 'NC':
+            if val == "NC":
                 message = "cannot be NC"
-                invalid_items.append({'key': key, 'val': val, 'message': message})
+                invalid_items.append(
+                    {"key": key, "val": val, "message": message}
+                )
     return invalid_items
 
-def legacy_assignment_check(pinName_content):
+
+def legacy_assignment_check(pin_name_content):
     invalid_items = []
-    legacy_assignments = dict(re.findall("^\s*((?:LED|BUTTON)\d*)\s*=\s*([a-zA-Z0-9_]+)", pinName_content, re.MULTILINE))
+    legacy_assignments = dict(
+        re.findall(
+            "^\s*((?:LED|BUTTON)\d*)\s*=\s*([a-zA-Z0-9_]+)",
+            pin_name_content,
+            re.MULTILINE,
+        )
+    )
     for key, val in legacy_assignments.items():
         message = "legacy assignment; LEDs and BUTTONs must be #define'd"
-        invalid_items.append({'key': key, 'val': val, 'message': message})
+        invalid_items.append({"key": key, "val": val, "message": message})
     return invalid_items
 
 
 def print_summary(report):
     targets = []
+
     for case in report:
-        targets.append(case['platform_name'])
-    targets = list(dict.fromkeys(targets))
-    
+        targets.append(case["platform_name"])
+
+    # TODO: remove duplicate
+    target_names = list(dict.fromkeys(targets))
+
     table = []
-    for target in targets:
-        result = 'PASSED'
+    for target_name in target_names:
         error_count = 0
+
         for case in report:
-            if case['platform_name'] != target:
-                continue
-            if case['result'] == 'FAILED':
-                result = 'FAILED'
+            if (
+                case["platform_name"] == target_name
+                and case["result"] == "FAILED"
+            ):
                 error_count += 1
-        table.append([target, result, error_count])
-    
-    return tabulate(table, headers=['Platform name', 'Result', 'Error count'], tablefmt='grid')
+
+        table.append(
+            [target_name, "FAILED" if error_count else "PASSED", error_count]
+        )
+
+    return tabulate(
+        table,
+        headers=["Platform name", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
 
 def print_suite_summary(report):
     targets = []
     for case in report:
-        targets.append(case['platform_name'])
+        targets.append(case["platform_name"])
     targets = list(dict.fromkeys(targets))
-    
+
     table = []
     for target in targets:
         suites = []
         for case in report:
-            if case['platform_name'] != target:
+            if case["platform_name"] != target:
                 continue
-            suites.append(case['suite_name'])
+            suites.append(case["suite_name"])
         suites = list(dict.fromkeys(suites))
```

Comment:
```python
suites = {case["suite_name"] for case in report if case["suite_name"] == target}
```

You'll want to double check my logic at the end, but I think that's what you wanted.